### ISSUE TITLE
Mission conditions can't specify the bank as a point of origin

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#3297]
I am trying to write a "mission" that congratulates the player upon paying off their mortgage, like so:

```
mission "paid off mortgage"
	to offer
		not "unpaid mortgages"
	on offer
		log "Finally paid off the mortgage!"
		conversation
			`Though it seemed like an unimaginably large amount of money at the time, you have just paid off your mortgage. An earnest smile creeps across your face as it begins to dawn on you: you're free. No more daily payments. No more 2 AM messages from loan service officers. No more putting off your plans to buy a better ship.`
			`	Congratulations, captain. The galaxy awaits.`
				decline
```

Unfortunately, there's no way to be offered a mission at the bank, so I can't make the conversation pop up immediately upon paying off the balance. The two closest options are at the spaceport (after taking off and landing again), or on landing, but neither is really appropriate.

**Comment by Amazinite** [comment]
How about the ability to specify any location that the mission offers on a planet?
We already have these:
* nothing = spaceport
* `landing` = on landing
* `job` = job board

So why not add these?
* `bank`
* `outfitter`
* `shipyard`
* `hiring`
* `trading`

**Comment by tehhowch** [comment]
You could get the message to appear on takeoff the day after, via an invisible landing mission with a blank `on enter`.

One issue with the specific case you mention is that the mortgage can be paid off when without visiting the bank again, so the player might get the message long after the initial repayment if they don't take out new loans.

**Comment by ghoulavenger** [comment]
Oddly enough, if you're playing a true merchantman fleet, loans are actually your friends. When I do cargo fleets, I even take out loans to pay off loans (I'm not kidding, because doing so is basically refinancing your loan). So this congratulatory message you want seems a little off (since it seems to imply that having debt is a bad thing, when having debt can fuel expansion).

**Comment by flaviojs** [comment]
The current way of adding single words to the mission node is a pain to support in the long term.

For me it makes more sense to place something like that as a condition inside the `to offer` node. Illustrative syntax: `at <job/landing/assisting/boarding/bank/...>`
```
mission "paid off mortgage"
	to offer
		not "unpaid mortgages"
		at bank
```
 This also gives more freedom, since you can then mix it up with `and`/`or`.

**Comment by endless-sky** [comment]
Have you tested this? Landing missions ought to be offered in the bank, in addition to the landing "panel." (See [PlanetPanel](https://github.com/endless-sky/endless-sky/blob/master/source/PlanetPanel.cpp#L80).)

**Comment by Pointedstick** [comment]
Yes, I have. The problem is that I want the conversation to appear immediately once you pay off the mortgage. Making it a `landing` mission only shows it on the _next_ landing, not when you're already on the planet and in the bank.

**Comment by endless-sky** [comment]
That actually has nothing to do with the fact that you're in the bank - the issue is that the auto-conditions (like the total mortgage debt) get set when you land and don't get updated after that.

**Comment by Pointedstick** [comment]
Ah, okay.

**Comment by warp-core** [comment]
Did #5244 fix this?
